OrbitLinuxTracing/TracerThread.cpp

```diff
@@ -16,17 +16,25 @@ void TracerThread::Run(
   Reset();
 
   // perf_event_open refers to cores as "CPUs".
-  int32_t num_cpus = GetNumCores();
+  std::vector<int32_t> all_cpus;
+  for (int32_t cpu = 0; cpu < GetNumCores(); ++cpu) {
+    all_cpus.push_back(cpu);
+  }
+
+  std::vector<int32_t> cpuset_cpus = GetCpusetCpus(pid_);
+  if (cpuset_cpus.empty()) {
```

Comment:
Can this happen? Wouldn't this mean the process (which is already running) cannot run on any of the cores?

Reply:
At the moment, an empty cpuset_cpus indicates an error in reading the cpuset (I added logging the error), that's why it falls back to sampling all cores. 
I experimented a bit and trying to start a process with an empty cpuset fails with "cgroup change of group failed", hence a proper cpuset should not be empty (added a comment in GetCpusetCpus). That's why I left the logic unchanged.